### ISSUE TITLE
Working directory not respected by git-bash.exe

### ISSUE THREAD

**Issue by orgads** [git-for-windows/git#130]
Settings "Start in" for its shortcut has no effect.

**Comment by dscho** [comment]
> Settings "Start in" for its shortcut has no effect.

As it should be: the `Git Bash` explicitly switches the working directory to your home directory (which makes most sense as a default, much more than starting the `Git Bash` in `C:\Program Files\Git` which would be the default otherwise).

You can pass a `--cd=<directory>` command-line option, though.

Anyway, I really consider this *not* a bug, but I would consider it a bug if `Git Bash` would respect the working directory that is the default. Remember: `git-bash.exe` should be double-clickable *in a portable Git*. There is no "Settings 'Start in'" for `git-bash.exe` that we could set to "user's home, please".

**Comment by orgads** [comment]
Makes sense. Thanks for the fast response.

Is this flag documented anywhere?

**Comment by dscho** [comment]
> Is this flag documented anywhere?

No, not yet. If you could take care of that, and of the `--no-cd` flag, I would be very thankful.

**Comment by orgads** [comment]
Where does it belong?

**Comment by dscho** [comment]
Ah. That is the question, isn't it? Let's rephrase it a little bit, though: Where would this information have crossed *your* eyes without having to search for it?

**Comment by orgads** [comment]
Release notes?

**Comment by dscho** [comment]
Sure!

**Comment by sukiletxe** [comment]
Hello,
I had the problem reported in the issue, and looked in the release notes just after being pointed out to them. Do you think it's worth it to document the `--cd=` and the `--no-cd` (I do)? If so, I'll open a pull request.
Thanks.

**Comment by dscho** [comment]
@sukiletxe excellent idea, go for it!

**Comment by sukiletxe** [comment]
I can't find the file to edit (the readme.portable shipped with the portable archive). I tried to search using some keywords and had no luck. Is the readme.portable added only in the release process? If so, and if I want to edit this file, should I put a copy in the repo?

**Comment by dscho** [comment]
> I can't find the file to edit (the readme.portable shipped with the portable archive).

It is here: https://github.com/git-for-windows/build-extra/blob/master/portable/root/README.portable

> Is the readme.portable added only in the release process?

Yes, and only as part of running https://github.com/git-for-windows/build-extra/blob/master/portable/release.sh

**Comment by sukiletxe** [comment]
Thanks for your response. That was completely my fault for not looking in the build-extra repo. Anyway, I've created a pull request, documenting --cd, --no-cd and HOME.

**Comment by dscho** [comment]
FWIW I made up my mind about `git-bash.exe` *always* switching directory and determined that my original reasoning was flawed: I wanted `git-bash.exe` in a portable Git to behave like the Git Bash start menu item of the non-portable version. But that is usually not what is needed! A most typical use case is for e.g. GitHub for Windows to start a Git Bash in a given working directory.

So I changed the behavior of `git-bash.exe` to require the new `--cd-to-home` option to behave like the Git Bash start menu item (which uses that option now). @sukiletxe may I ask you to adjust the README once more?
